index.src.html

```diff
@@ -958,6 +960,103 @@ <h3 id="strip-url">
     </li>
   </ol>
 
+  <h3 id="ancestor-origin">
+    Determine the Ancestor Origin availble to a Location
+  </h3>
+  Given a <a>Location</a> <var>location</var> and a <a>browsing context</a> <var>context</var>
```

Comment:
It seems simplest to have this behave as much like the rest of referrer policy as possible, including respecting the secure context boundary in a similar manner.  

The existing algorithms in Referrer Policy use an environment settings object, Request's client.  

The existing algorithm in HTML uses a browsing context.  

It does look like the necessary pieces of information are available in a Document, as well.  Although I think for consistency we would need to use Document's Location's URL, not just origin, to get proper secure context testing for blob: and filesystem: URLs. 

Otherwise I honestly don't know what the reasons are to prefer Document vs. browsing context vs. environment settings object since the same relevant input data seems to be reachable from each.